src/server/game/Handlers/TicketHandler.cpp

```diff
@@ -42,140 +42,101 @@ void WorldSession::HandleGMTicketCreateOpcode(WorldPacket& recvData)
     }
 
     GMTicketResponse response = GMTICKET_RESPONSE_CREATE_ERROR;
-    GmTicket* ticket = sTicketMgr->GetTicketByPlayer(GetPlayer()->GetGUID());
+    GmTicket* ticket = sSupportMgr->GetGmTicketByPlayerGuid(GetPlayer()->GetGUID());
 
     if (ticket && ticket->IsCompleted())
-        sTicketMgr->CloseTicket(ticket->GetId(), GetPlayer()->GetGUID());
+        sSupportMgr->CloseTicket<GmTicket>(ticket->GetId(), GetPlayer()->GetGUID());
 
     // Player must not have ticket
     if (!ticket || ticket->IsClosed())
     {
-        uint32 mapId;
-        float x, y, z;
-        std::string message;
-        uint32 needResponse;
-        bool needMoreHelp;
-        uint32 count;
-        std::list<uint32> times;
-        uint32 decompressedSize;
         std::string chatLog;
 
-        recvData >> mapId;
-        recvData >> x >> y >> z;
-        recvData >> message;
-
-        recvData >> needResponse;
-        recvData >> needMoreHelp;
-
-        recvData >> count;
-
-        for (uint32 i = 0; i < count; i++)
+        if (packet.DataLength > 0 && packet.ChatHistoryData.DecompressedSize < 0xFFFF)
         {
-            uint32 time;
-            recvData >> time;
-            times.push_back(time);
-        }
-
-        recvData >> decompressedSize;
-
-        if (count && decompressedSize && decompressedSize < 0xFFFF)
-        {
-            uint32 pos = recvData.rpos();
             ByteBuffer dest;
-            dest.resize(decompressedSize);
+            dest.resize(size_t(packet.ChatHistoryData.DecompressedSize));
 
-            uLongf realSize = decompressedSize;
-            if (uncompress(dest.contents(), &realSize, recvData.contents() + pos, recvData.size() - pos) == Z_OK)
-            {
+            uLongf realSize = packet.ChatHistoryData.DecompressedSize;
```

Comment:
not sure if we want to use uLongf type since we usually use only uint32/64/size_t